src/main/java/uk/gov/ida/verifyserviceprovider/mappers/MatchingDatasetToNonMatchingAttributesMapper.java

```diff
@@ -0,0 +1,101 @@
+            .map(MatchingDatasetToNonMatchingAttributesMapper::convertWrappedJodaLocalDateToJavaLocalDate)
+            .findFirst();
+
+        Optional<NonMatchingVerifiableAttribute<String>> firstName = firstNameValue.map(this::mapToNonMatchingVerifiableAttribute);
+        List<NonMatchingVerifiableAttribute<String>> middleNames = matchingDataset.getMiddleNames().stream().map(this::mapToNonMatchingVerifiableAttribute).collect(Collectors.toList());
+        List<NonMatchingVerifiableAttribute<String>> surnames = matchingDataset.getSurnames().stream().map(this::mapToNonMatchingVerifiableAttribute).collect(Collectors.toList());
+        Optional<NonMatchingVerifiableAttribute<LocalDate>> dateOfBirth = birthDateValue.map(this::mapToNonMatchingVerifiableAttribute);
+        Optional<Gender> gender = matchingDataset.getGender().map(SimpleMdsValue::getValue);
+        List<NonMatchingVerifiableAttribute<NonMatchingAddress>> addresses = mapAddresses(matchingDataset.getAddresses());
+
+        return new NonMatchingAttributes(
+            firstName,
+            middleNames,
+            surnames,
+            dateOfBirth,
+            gender,
+            addresses
+        );
+    }
+
+    private <T> NonMatchingVerifiableAttribute<T> mapToNonMatchingVerifiableAttribute(SimpleMdsValue<T> simpleMdsValueOptional) {
+        Optional<LocalDateTime> from = Optional.of(convertJodaDateTimeToJavaLocalDateTime(simpleMdsValueOptional.getFrom()));
+        Optional<LocalDateTime> to = Optional.of(convertJodaDateTimeToJavaLocalDateTime(simpleMdsValueOptional.getTo()));
+
+        return new NonMatchingVerifiableAttribute<>(
+            simpleMdsValueOptional.getValue(),
+            simpleMdsValueOptional.isVerified(),
+            from,
+            to
+        );
+    }
+
+    private List<NonMatchingVerifiableAttribute<NonMatchingAddress>> mapAddresses(List<Address> addresses) {
+        List<NonMatchingVerifiableAttribute<NonMatchingAddress>> output = new java.util.ArrayList<>();
+
+        for (Address input : addresses) {
+            NonMatchingAddress transformedAddress = new NonMatchingAddress(
+                input.getLines(),
+                input.getPostCode().orElse(""),
+                input.getInternationalPostCode().orElse("")
+            );
+
+            Optional<LocalDateTime> from = Optional.ofNullable(input.getFrom()).map(MatchingDatasetToNonMatchingAttributesMapper::convertJodaDateTimeToJavaLocalDateTime);
+            Optional<LocalDateTime> to = input.getTo().map(MatchingDatasetToNonMatchingAttributesMapper::convertJodaDateTimeToJavaLocalDateTime);
+
+            NonMatchingVerifiableAttribute<NonMatchingAddress> addressAttribute = new NonMatchingVerifiableAttribute<>(
+                transformedAddress,
+                input.isVerified(),
+                from,
+                to
+            );
+
+            output.add(addressAttribute);
+        }
+
+        return output;
+    }
+
```

Comment:
I promised Andy I'd get rid of the ugly date mapping code below and instead just keep using the Joda dates but make sure they get serialised in the right format; in the end I for forgot (and ran out of time/patience), but it would be worth making this enhancement later.  I've added it to my list of tech debt (which I'll be adding to the tech debt Trello properly, later today).